src/EventStore.Common/Utils/ContainerizedEnvironment.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+
+namespace EventStore.Common.Utils;
+
+public static class ContainerizedEnvironment {
+
+	public const int ReaderThreadCount = 4;
+	public const int WorkerThreadCount = 5;
+	public const int StreamInfoCacheCapacity = 100000;
+
+	private static bool _isContainer;
+
+	public static bool IsRunningInContainer() {
+		string dotnetContainerEnv = Environment.GetEnvironmentVariable("DOTNET_RUNNING_IN_CONTAINER");
+		_isContainer = !string.IsNullOrEmpty(dotnetContainerEnv);
+		return _isContainer;
+	}
+
+	public static bool IsContainer() {
```

Comment:
there is a danger here that if `IsContainer` is called before `IsRunningInContainer` then `_isContainer` won't be populated